---
layout: page
permalink: /publications/
title: Publications
description: Peer-reviewed papers and pre-prints with links to code and data
---

- My page on [Google Scholar](https://scholar.google.ca/citations?user=c83d8tgAAAAJ).
- Peer review notice: in addition to peer-reviewed journals, I publish
  papers at machine learning conferences such as ICML and NIPS, which
  have double-blind peer reviews, and only accept about 20% of
  submitted papers.
- Reproducible Research notice: a "Reproducible" link for each paper below
  provides source code and data that I used to perform the analysis and make
  the figures/tables.
- Software notice: a "Software" link for each paper below provides a software
  package with a reference implementation of the algorithms described
  in our paper.

### In progress

- Drouin A, Hocking TD, Laviolette F. Max margin interval trees. Under
  review. [Software](https://github.com/aldro61/mmit),
  [Reproducible](https://github.com/tdhock/mmit-paper)
- Sievert C, Cai J, VanderPlas S, Khan F, Hocking TD. Extending
  ggplot2's grammar of graphics implementation for linked and dynamic
  graphics on the web. [Software](https://github.com/tdhock/animint),
  [Reproducible](https://github.com/tdhock/animint-paper)
- Hocking TD, Khare A. Learning penalty functions for changepoint
  detection using elastic-net regularized accelerated failure time
  models. [Software](https://github.com/anujkhare/iregnet),
  [Reproducible](https://github.com/tdhock/iregnet-paper)
- Narahara M, Hocking TD, Bourque G, Yamada R, Setoh K, Matsuda F,
  Lathrop M. Transcriptomic analysis of antibody responses to seasonal
  influenza vaccine reveals predictive gene signatures and potential
  key transcription factors.
- Alirezaie N, Majewski J, Hocking TD. A supervised machine
  learning method for predicting pathogenicity of genetic variants.

### 2017

- Hocking TD, Rigaill G, Fearnhead P, Bourque G. A log-linear
  segmentation algorithm for peak detection in genomic
  data. [Preprint arXiv:1703.03352](https://arxiv.org/abs/1703.03352). Under
  review at Annals of Applied
  Statistics. [Software](https://github.com/tdhock/coseg),
  [Reproducible](https://github.com/tdhock/PeakSegFPOP-paper)
- Hocking TD, Goerner-Potvin P, Morin A, Shao X, Pastinen T, Bourque
  G. Optimizing ChIP-seq peak detectors using visual labels and
  supervised machine learning.  Bioinformatics (2017) 33 (4):
  491-499. [pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27797775),
  [Software](https://github.com/tdhock/PeakError),
  [Reproducible](https://bitbucket.org/mugqic/chip-seq-paper)

### 2016

- Shimada K, Shimada S, Sugimoto K, Nakatochi M, Suguro M, Hirakawa A,
  Hocking TD, Takeuchi I, Tokunaga T, Takagi Y, Sakamoto A, Aoki T,
  Naoe T, Nakamura S, Hayakawa F, Seto M, Tomita A, Kiyoi
  H. Development and analysis of patient-derived xenograft mouse
  models in intravascular large B-cell lymphoma.
  Leukemia 2016. [pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27001523)
- Chicard M, Boyault S, Colmet-Daage L, Richer W, Gentien D, Pierron
  G, Lapouble E, Bellini A, Clement N, Iacono I, Br√©jon S, Carrere M,
  Reyes C, Hocking TD, Bernard V, Peuchmaur M, Corradini N,
  Faure-Conter C, Coze C, Plantaz D, Defachelles A-S, Thebaud E,
  Gambart M, Millot F, Valteau-Couanet D, Michon J, Puisieux A,
  Delattre O, Combaret V, Schleiermacher G. Genomic copy number
  profiling using circulating free tumor DNA highlights heterogeneity
  in neuroblastoma. Clinical Cancer
  Research 2016. [journal](http://clincancerres.aacrjournals.org/content/early/2016/11/03/1078-0432.CCR-16-0500)
- Maidstone R, Hocking TD, Rigaill G, Fearnhead P. On optimal multiple
  changepoint algorithms for large data.  Statistics and Computing
  (2016). doi:10.1007/s11222-016-9636-3
  [journal](https://link.springer.com/article/10.1007/s11222-016-9636-3),
  [Software](https://r-forge.r-project.org/R/?group_id%3D1851),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/benchmark/?root%3Dopfp)

### 2015

- Hocking TD, Rigaill G, Bourque G. PeakSeg: constrained optimal
  segmentation and supervised penalty learning for peak detection in
  count data.
  International Conference on Machine Learning (ICML), 2015. 
  [PMLR](http://proceedings.mlr.press/v37/hocking15.html), 
  [video](http://videolectures.net/icml2015_hocking_count_data/?q=hocking),
  [Software](https://github.com/tdhock/PeakSegDP),
  [Reproducible](https://github.com/tdhock/PeakSeg-paper)
- Hocking TD. A breakpoint detection error function for segmentation
  model selection and
  validation. [Preprint arXiv:1509.00368](https://arxiv.org/abs/1509.00368),
  [Software](http://r-forge.r-project.org/scm/?group_id=1540),
  [Reproducible](https://github.com/tdhock/breakpointError-orig)
- Hocking TD, Bourque G. PeakSegJoint: fast supervised peak detection
  via joint segmentation of multiple count data
  samples. [Preprint arXiv:1506.01286](https://arxiv.org/abs/1509.00368),
  [Software](https://github.com/tdhock/PeakSegJoint),
  [Reproducible](https://github.com/tdhock/PeakSegJoint-paper)

### 2014

- Suguro M, Yoshida N, Umino A, Kato H, Tagawa H, Nakagawa M, Fukuhara
  N, Karnan S, Takeuchi I, Hocking TD, Arita K, Karube K, Tsuzuki S,
  Nakamura S, Kinoshita T, Seto M. Clonal heterogeneity of lymphoid
  malignancies correlates with poor prognosis.  Cancer Sci. 2014
  Jul;105(7):897-904. [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24815991)
- Hocking TD, Boeva V, Rigaill G, Schleiermacher G, Janoueix-Lerosey
  I, Delattre O, Richer W, Bourdeaut F, Suguro M, Seto M, Bach F, Vert
  J-P. SegAnnDB: interactive Web-based genomic segmentation.
  Bioinformatics (2014) 30 (11):
  1539-1546. DOI:10.1093/bioinformatics/btu072
  [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24493034),
  [Software](https://github.com/tdhock/SegAnnDB),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/breakpoints/webapp/applications-note/)
- Venuto D, Spanurattana S, Sugiyama M, Hocking TD. Support vector
  comparison
  machines. [Preprint arXiv:1401.8008](https://arxiv.org/abs/1401.8008).
  [Software](https://github.com/tdhock/rankSVMcompare),
  [Reproducible](https://github.com/tdhock/compare-paper)

### 2013

- Hocking TD, Wutzler T, Ponting K and Grosjean P. Sustainable,
  extensible documentation generation using inlinedocs.  Journal of
  Statistical Software (2013), 54(6), 1-20. DOI:10.18637/jss.v054.i06
  [journal](https://www.jstatsoft.org/article/view/v054i06),
  [Software](https://cran.r-project.org/package=inlinedocs),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/jss762/?root=inlinedocs)
- Hocking TD, Schleiermacher G, Janoueix-Lerosey I, Boeva V, Cappo J,
  Delattre O, Bach F, Vert J-P. Learning smoothing models of copy
  number profiles using breakpoint annotations.  BMC Bioinfo. 2013,
  14:164. DOI:10.1186/1471-2105-14-164
  [journal](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-164),
  [Software](https://cran.r-project.org/src/contrib/Archive/bams/),
  [Reproducible](http://cbio.ensmp.fr/~thocking/papers/2012-01-27-Breakpoint-annotation-model-smoothing/HOCKING-breakpoint-annotation-model-smoothing.tgz)
- Hocking TD, Rigaill G, Bach F, Vert J-P. Learning sparse penalties
  for change-point detection using max-margin interval regression.
  International Conference on Machine Learning
  (ICML), 2013. [PMLR](http://proceedings.mlr.press/v28/hocking13.html),
  [video](http://techtalks.tv/talks/learning-sparse-penalties-for-change-point-detection-using-max-margin-interval-regression/58208/)
  [Software](https://github.com/tdhock/penaltyLearning),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/?root=breakpoints)

### 2012

- Hocking TD, Rigaill G. SegAnnot: an R package for fast segmentation
  of annotated piecewise constant signals,
  [Preprint hal-00759129](https://hal.inria.fr/hal-00759129). 
  [Software](https://r-forge.r-project.org/scm/?group_id=1541),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/SegAnnot/inst/doc/?root=breakpoints)
- Hocking TD. Learning algorithms and statistical software, with
  applications to bioinformatics. PhD thesis, Ecole Normale Superieure
  de Cachan,
  France. [tel-00906029](https://tel.archives-ouvertes.fr/tel-00906029/), 
  [Reproducible](http://cbio.ensmp.fr/~thocking/papers/2012-11-20-PhD-thesis/HOCKING-phd-thesis.tgz)

### 2011

- Hocking TD, Joulin A, Bach F, Vert J-P. Clusterpath: an Algorithm
  for Clustering using Convex Fusion Penalties.  International
  Conference on Machine Learning
  (ICML), 2011. [pdf](http://www.icml-2011.org/papers/419_icmlpaper.pdf),
  [video](http://techtalks.tv/talks/clusterpath-an-algorithm-for-clustering-using-convex-fusion-penalties/54405/)
  [Software](https://r-forge.r-project.org/scm/?group_id=1090),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/?root=clusterpath)

### 2010

- Gautier M, Hocking TD, Foulley JL. A Bayesian outlier criterion to
  detect SNPs under selection in large data sets.  PloS ONE 5 (8),
  e11913
  (2010). [journal](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0011913)

### 2008

- Doyon Y, McCammon JM, Miller JC, Faraji F, Ngo C, Katibah GE, Amora
  R, Hocking TD, Zhang L, Rebar EJ, Gregory PD, Urnov FD, Amacher
  SL. Heritable targeted gene disruption in zebrafish using designed
  zinc-finger nucleases.  Nature biotechnology 26 (6), 702-70
  (2008). [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/18500334)



