\documentclass[margin,line]{res}
\usepackage{verbatim}
\usepackage{url}
\urlstyle{same}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}

\oddsidemargin -.5in
\evensidemargin -.5in
\textwidth=6.0in
\itemsep=0in
\parsep=0in

\newenvironment{list1}{
  \begin{list}{\ding{113}}{%
      \setlength{\itemsep}{0in}
      \setlength{\parsep}{0in} \setlength{\parskip}{0in}
      \setlength{\topsep}{0in} \setlength{\partopsep}{0in} 
      \setlength{\leftmargin}{0.17in}}}{\end{list}}
\newenvironment{list2}{
  \begin{list}{$\bullet$}{%
      \setlength{\itemsep}{0in}
      \setlength{\parsep}{0in} \setlength{\parskip}{0in}
      \setlength{\topsep}{0in} \setlength{\partopsep}{0in} 
      \setlength{\leftmargin}{0.2in}}}{\end{list}}

\begin{document}

\name{
\begin{tabular*}{7.4in} {@{\extracolsep{\fill}}lr}
Toby Dylan Hocking & Curriculum Vitae 
\end{tabular*}
}

\begin{resume}
\section{\sc Name and Affiliation}
\vspace{.05in}
\begin{tabular*}{6.1in} {@{\extracolsep{\fill}}ll}
  Toby Dylan Hocking& Citizenship: USA \\            
 Research Associate  & Phone:  (514) 398-3311 x 00721 \\   
McGill University & E-mail:  toby.hocking@mail.mcgill.ca       \\
Department of Human Genetics &  Webpage: \url{http://tdhock.github.io}\\     
\end{tabular*}


\section{\sc Work Experience}

{\bf McGill University}, Montreal, Canada (2014-present).\\
\vspace*{-.1in}
\begin{list2}
\item[] Postdoc with Guillaume Bourque, Department of Human Genetics.
\item[]``Changepoint detection and regression models for peak detection in genomic data.''
\end{list2}

{\bf Tokyo Institute of Technology}, Tokyo, Japan (2013).\\
\vspace*{-.1in}
\begin{list2}
\item[] Postdoc with Masashi Sugiyama, Department of Computer Science.
\item[] ``Support vector machines for ranking and comparing.''
\end{list2}

{\bf Sangamo BioSciences}, Richmond, CA, USA (2006-2008).\\
\vspace*{-.1in}
\begin{list2}
\item[] Research Assistant with Jeff Miller in the Technology group.
\item[] ``A web app for visualization and statistical analysis of experimental data.''
\end{list2}

\section{\sc Academic and Training Background}

{\bf \'{E}cole Normale Sup\'{e}rieure}, Cachan, France (2009-2012).\\
\vspace*{-.1in}
\begin{list2}
\item[] Ph.D. with Francis Bach, D\'{e}partement d'Informatique; Jean-Philippe Vert, Institut Curie.
\item[] ``Learning algorithms and statistical software, with applications to bioinformatics."
\end{list2}

{\bf Universit\'e Paris 6}, Paris, France (2008-2009).\\
\vspace*{-.1in}
\begin{list2}
\item[] Master of Statistics, internship at INRA with Mathieu Gautier and Jean-Louis Foulley.
\item[] ``A Bayesian Outlier Criterion to Detect SNPs under Selection in Large Data Sets."
\end{list2}

{\bf University of California, Berkeley}, CA, USA (2002-2006).\\
\vspace*{-.1in}
\begin{list2}
\item[] Double B.A. in Statistics, Molecular and Cell Biology; thesis in Statistics with Terry Speed.
\item[] ``Chromosomal copy number analysis using SNP microarrays and a binomial test statistic.'' 
\end{list2}

\section{\sc Selected Accomplishments and Honours}

``Mobilit\'e entrant'' travel award to work with Guillem Rigaill in
Universit\'e Evry, France, 2016.

International useR conference, Best Student Poster Award, ``Adding
direct labels to plots,'' 2011.

INRIA/INRA (French computer science and agricultural research institutes), Ph.D. scholarship, 2009 (declined).

UC Berkeley, Department of Statistics VIGRE research scholarship, 2001.

UC Berkeley, Cal Band George Miller scholarship, 2000.

\section{\sc Selected Peer-Rfeviewed Publications}

In addition to peer-reviewed journals, I publish papers at
machine learning conferences such as ICML and NIPS, which have
double-blind peer reviews, and only accept about 20\% of submitted
papers.

Drouin A, {\bf Hocking TD}, Laviolette F. Max margin interval
trees. {\it Neural Information Processing Systems (NIPS)}, 2017.

{\bf Hocking TD}, Goerner-Potvin P, Morin A, Shao X, Pastinen T,
Bourque G. Optimizing ChIP-seq peak detectors using visual labels and
supervised machine learning. {\it Bioinformatics} (2017) 33 (4): 491-499.

Shimada K, Shimada S, Sugimoto K, Nakatochi M, Suguro M, Hirakawa A,
{\bf Hocking TD}, Takeuchi I, Tokunaga T, Takagi Y, Sakamoto A, Aoki T, Naoe
T, Nakamura S, Hayakawa F, Seto M, Tomita A, Kiyoi H. Development and
analysis of patient-derived xenograft mouse models in intravascular
large B-cell lymphoma. {\it Leukemia} 2016.

Chicard M, Boyault S, Colmet-Daage L, Richer W, Gentien D, Pierron G,
Lapouble E, Bellini A, Clement N, Iacono I, Br√©jon S, Carrere M, Reyes
C, {\bf Hocking TD}, Bernard V, Peuchmaur M, Corradini N, Faure-Conter
C, Coze C, Plantaz D, Defachelles A-S, Thebaud E, Gambart M, Millot F,
Valteau-Couanet D, Michon J, Puisieux A, Delattre O, Combaret V,
Schleiermacher G. Genomic copy number profiling using circulating free
tumor DNA highlights heterogeneity in neuroblastoma. {\it Clinical Cancer
Research} 2016.

Maidstone R, {\bf Hocking TD}, Rigaill G, Fearnhead P. On optimal
multiple changepoint algorithms for large data. {\it Statistics and
Computing} (2016). doi:10.1007/s11222-016-9636-3 

{\bf Hocking TD}, Rigaill G, Bourque G. PeakSeg: constrained optimal
segmentation and supervised penalty learning for peak detection in
count data. {\it International Conference on Machine Learning (ICML)},
2015.

Suguro M, Yoshida N, Umino A, Kato H, Tagawa H, Nakagawa M, Fukuhara
N, Karnan S, Takeuchi I, {\bf Hocking TD}, Arita K, Karube K, Tsuzuki
S, Nakamura S, Kinoshita T, Seto M. Clonal heterogeneity of lymphoid
malignancies correlates with poor prognosis. {\it Cancer Sci.} 2014
Jul;105(7):897-904.

{\bf Hocking TD}, Boeva V, Rigaill G, Schleiermacher G,
Janoueix-Lerosey I, Delattre O, Richer W, Bourdeaut F, Suguro M, Seto
M, Bach F, Vert J-P. SegAnnDB: interactive Web-based genomic
segmentation. {\it Bioinformatics} (2014) 30 (11):
1539-1546. DOI:10.1093/bioinformatics/btu072

{\bf Hocking TD}, Wutzler T, Ponting K and Grosjean P. Sustainable,
extensible documentation generation using inlinedocs. {\it Journal of
Statistical Software} (2013), 54(6), 1-20. DOI:10.18637/jss.v054.i06

{\bf Hocking TD}, Schleiermacher G, Janoueix-Lerosey I, Boeva V, Cappo
J, Delattre O, Bach F, Vert J-P. Learning smoothing models of copy
number profiles using breakpoint annotations. {\it BMC Bioinfo.} 2013,
14:164. DOI:10.1186/1471-2105-14-164

{\bf Hocking TD}, Rigaill G, Bach F, Vert J-P. Learning sparse
penalties for change-point detection using max-margin interval
regression. {\it International Conference on Machine Learning (ICML)}, 2013.

{\bf Hocking TD}, Joulin A, Bach F, Vert J-P. Clusterpath: an
Algorithm for Clustering using Convex Fusion Penalties. {\it International Conference on Machine Learning (ICML)}, 2011.

Gautier M, {\bf Hocking TD}, Foulley JL. A Bayesian outlier criterion
to detect SNPs under selection in large data sets. {\it PloS ONE} 5
(8), e11913 (2010).

Doyon Y, McCammon JM, Miller JC, Faraji F, Ngo C, Katibah GE, Amora R,
{\bf Hocking TD}, Zhang L, Rebar EJ, Gregory PD, Urnov FD, Amacher
SL. Heritable targeted gene disruption in zebrafish using designed
zinc-finger nucleases. {\it Nature biotechnology} 26 (6), 702-70
(2008).

{\bf Hocking TD}, Rigaill G, Fearnhead P, Bourque G. A log-linear
segmentation algorithm for peak detection in genomic data. Preprint
arXiv:1703.03352. Under review at {\it Annals of Applied Statistics}.

Sievert C, Cai J, VanderPlas S, Khan F, Ferris K, {\bf Hocking
  TD}. Extending ggplot2 for linked and dynamic web graphics. Under
review at {\it Journal of Computational and Graphical Statistics}.

Venuto D, {\bf Hocking TD}, Spanurattana S, Sugiyama M. Support vector
comparison machines. Preprint arXiv:1401.8008. Under review at {\it
  Journal of Machine Learning Research}.

\section{Selected Patent Applications and Patents}

None.

\vskip 2cm

\section{Selected Research Funding}

None.

\end{resume}
\end{document}




