\documentclass[margin,line]{res}
\usepackage{verbatim}
\usepackage{url}
\urlstyle{same}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}

\oddsidemargin -.5in
\evensidemargin -.5in
\textwidth=6.0in
\itemsep=0in
\parsep=0in

\newenvironment{list1}{
  \begin{list}{\ding{113}}{%
      \setlength{\itemsep}{0in}
      \setlength{\parsep}{0in} \setlength{\parskip}{0in}
      \setlength{\topsep}{0in} \setlength{\partopsep}{0in} 
      \setlength{\leftmargin}{0.17in}}}{\end{list}}
\newenvironment{list2}{
  \begin{list}{$\bullet$}{%
      \setlength{\itemsep}{0in}
      \setlength{\parsep}{0in} \setlength{\parskip}{0in}
      \setlength{\topsep}{0in} \setlength{\partopsep}{0in} 
      \setlength{\leftmargin}{0.2in}}}{\end{list}}

\begin{document}

\name{
\begin{tabular*}{7.4in} {@{\extracolsep{\fill}}lr}
Toby Dylan Hocking & Curriculum Vitae 
\end{tabular*}
}

\begin{resume}
\section{\sc Contact and General Info}
\vspace{.05in}
\begin{tabular*}{6.1in} {@{\extracolsep{\fill}}ll}
 NAU Building 90, Office 210 & Birth: 17 March 1984 in Newport Beach, California\\
 1295 S. Knoles Dr.  & Citizenship: USA \\            
  Flagstaff, AZ 86011 & Language skills:\\
  Web: \url{http://tdhock.github.io} & English (native)\\
  E-mail:  toby.hocking@nau.edu & French (fluent since 2009)\\
\end{tabular*}

\section{\sc Research Interests}

Statistical software, data visualization techniques, and machine
learning models. Emphasis on fast algorithms for large datasets, based
on constrained optimization algorithms (regression, classification,
ranking, clustering, segmentation, changepoint detection, survival
analysis). Applications to genomics, neuroscience, audio, internet,
sensors, recommendation systems.

\section{\sc Professional Experience}

{\bf Northern Arizona University}, Flagstaff, Arizona, USA (2018-present).\\
\vspace*{-.1in}
\begin{list2}
\item[] Assistant Professor, School of Informatics, Computing, and Cyber Systems.
\item[] ``Optimization algorithms for machine learning and interactive data analysis.''
\end{list2}

{\bf McGill University}, Montreal, Canada (2014-2018).\\
\vspace*{-.1in}
\begin{list2}
\item[] Postdoc with Guillaume Bourque, Department of Human Genetics.
\item[]``Changepoint detection and regression models for peak detection in genomic data.''
\end{list2}

{\bf Tokyo Institute of Technology}, Tokyo, Japan (2013).\\
\vspace*{-.1in}
\begin{list2}
\item[] Postdoc with Masashi Sugiyama, Department of Computer Science.
\item[] ``Support vector machines for ranking and comparing.''
\end{list2}

{\bf Sangamo BioSciences}, Richmond, CA, USA (2006-2008).\\
\vspace*{-.1in}
\begin{list2}
\item[] Research Assistant with Jeff Miller in the Technology group.
\item[] ``A web app for visualization and statistical analysis of experimental data.''
\end{list2}

\section{\sc Education}

{\bf \'{E}cole Normale Sup\'{e}rieure}, Cachan, France (2009-2012).\\
\vspace*{-.1in}
\begin{list2}
\item[] Ph.D. with Francis Bach, D\'{e}partement d'Informatique; Jean-Philippe Vert, Institut Curie.
\item[] ``Learning algorithms and statistical software, with applications to bioinformatics."
\end{list2}

{\bf Universit\'e Paris 6}, Paris, France (2008-2009).\\
\vspace*{-.1in}
\begin{list2}
\item[] Master of Statistics, internship at INRA with Mathieu Gautier and Jean-Louis Foulley.
\item[] ``A Bayesian Outlier Criterion to Detect SNPs under Selection in Large Data Sets."
\end{list2}

{\bf University of California, Berkeley}, CA, USA (2002-2006).\\
\vspace*{-.1in}
\begin{list2}
\item[] Double B.A. in Statistics, Molecular and Cell Biology; thesis in Statistics with Terry Speed.
\item[] ``Chromosomal copy number analysis using SNP microarrays and a binomial test statistic.'' 
\end{list2}

\section{\sc Honors and Awards (Selected)}

R Consortium Grant, \$34,000, Jan-Dec 2020, ``RcppDeepState: an easy
way to fuzz test compiled code in R packages.''

``Mobilit\'e entrant'' travel award to work with Guillem Rigaill in
Universit\'e Evry, France, 2016.

International useR conference, Best Student Poster Award, ``Adding
direct labels to plots,'' 2011.

INRIA/INRA (French computer science and agricultural research institutes), Ph.D. scholarship, 2009 (declined).

UC Berkeley, Department of Statistics VIGRE research scholarship, 2001.

UC Berkeley, Cal Band George Miller scholarship, 2000.

\section{\sc Papers in progress and under review}

Liehrmann A, {\bf Hocking TD}. Optimal multiple changepoint detection
models for recognition of histone modification events in ChIP-Seq
data.

Barnwal A, Cho H, {\bf Hocking TD}. Survival regression with accelerated
failure time model in XGBoost.

{\bf Hocking TD}, Khare A. Learning penalty functions for changepoint
detection using elastic-net regularized accelerated failure time
models.

Chaves AP, Egbert J, {\bf Hocking TD}, Doerry E, Gerosa MA. Chatbots
language design: the influence of language use on user
experience. Under review at ACM Transactions on Computer-Human
Interaction.

Abraham A, Prys-Jones T, De Cuyper A, Ridenour C, Hempson G, {\bf
  Hocking TD}, Clauss M, Doughty C. Improved estimation of gut passage
time considerably affects trait-based dispersal models. Under review
at {\it Functional Ecology}.

{\bf Hocking TD}, Srivastava A. Labeled Optimal Partitioning. Preprint
arXiv:2006.13967.

Fotoohinasab A, {\bf Hocking TD}, Afghah F. A Greedy Graph Search
Algorithm Based on Changepoint Analysis for Automatic QRS-Complex
Detection. Under review at {\it Computers in Biology and Medicine},
arXiv:2004.13558.

{\bf Hocking TD}, Vargovich J. Linear time dynamic programming for the
exact path of optimal models selected from a finite set. Under review
at {\it Journal of Computational and Graphical Statistics},
arXiv:2003.02808.

Runge V, {\bf Hocking TD}, Romano G, Afghah F, Fearnhead P, Rigaill
G. gfpop: an R Package for Univariate Graph-Constrained Change-point
Detection. Under review at {\it Journal of Statistical Software},
arXiv:2002.03646.

{\bf Hocking TD}. Regular expressions and reshaping using data tables
and the nc package. Under review at R Journal.

Venuto D, {\bf Hocking TD}, Spanurattana S, Sugiyama M. Support vector
comparison machines. Under review at {\it Machine Learning},
arXiv:1401.8008.

\section{\sc Peer-reviewed publications}

In addition to peer-reviewed journals, I publish papers at
computer science conferences such as {\it ICML} and {\it NeurIPS}, with
double-blind peer reviews, and $\approx$20\% acceptance rates.

Fotoohinasab A, {\bf Hocking TD}, Afghah F. A Graph-Constrained Changepoint
Learning Approach for Automatic QRS-Complex Detection. Accepted for
{\it Asilomar Conference on Signals, Systems, and Computers},
arXiv:2004.13558.

{\bf Hocking TD}, Rigaill G, Fearnhead P, Bourque G. Generalized
Functional Pruning Optimal Partitioning (GFPOP) for Constrained
Changepoint Detection in Genomic Data. Accepted for publication in
{\it Journal of Statistical Software}, arXiv:1810.00117.

Fotoohinasab A, {\bf Hocking TD}, Afghah F. A Graph-constrained
Changepoint Detection Approach for ECG Segmentation. In proceedings of
{\it 42th Annual International Conference of the IEEE Engineering in
  Medicine and Biology Society (EMBC)}.

{\bf Hocking TD}, Rigaill G, Fearnhead P, Bourque G. Constrained
dynamic programming and supervised penalty learning algorithms for
peak detection in genomic data. {\it Journal of Machine Learning
  Research} 21(87):1--40, 2020.

{\bf Hocking TD}, Bourque G. Machine Learning Algorithms for
Simultaneous Supervised Detection of Peaks in Multiple Samples and
Cell Types. {\it Pacific Symposium on Biocomputing} 25:367-378.

{\bf Hocking TD}. Comparing namedCapture with other R packages for
regular expressions. {\it R Journal} (2019). doi:10.32614/RJ-2019-050

Jewell S, {\bf Hocking TD}, Fearnhead P, Witten D. Fast Nonconvex
Deconvolution of Calcium Imaging Data. {\it Biostatistics} (2019), doi:
10.1093/biostatistics/kxy083.

Depuydt P, Koster J, Boeva V, {\bf Hocking TD}, Speleman F,
Schleiermacher G, De Preter K. Meta-mining of copy number profiles of
high-risk neuroblastoma tumors. {\it Scientific Data} (2018).

Alirezaie N, Kernohan KD, Hartley T, Majewski J, {\bf Hocking
  TD}. ClinPred: Prediction Tool to Identify Disease-Relevant
Nonsynonymous Single-Nucleotide Variants. American Journal of Human
Genetics (2018). doi:10.1016/j.ajhg.2018.08.005

Sievert C, Cai J, VanderPlas S, Khan F, Ferris K, {\bf Hocking
  TD}. Extending ggplot2 for linked and dynamic web graphics. {\it
  Journal of Computational and Graphical Statistics} (2018).

Depuydt P, Boeva V, {\bf Hocking TD}, {\it et al}. Genomic Amplifications and Distal 6q
  Loss: Novel Markers for Poor Survival in High-risk Neuroblastoma
  Patients. Journal of the National Cancer
  Institute (2018). DOI:10.1093/jnci/djy022.

Drouin A, {\bf Hocking TD}, Laviolette F. Max margin interval
trees. {\it Neural Information Processing Systems (NIPS)}, 2017.

{\bf Hocking TD}, Goerner-Potvin P, Morin A, Shao X, Pastinen T,
Bourque G. Optimizing ChIP-seq peak detectors using visual labels and
supervised machine learning. {\it Bioinformatics} (2017) 33 (4): 491-499.

Shimada K, Shimada S, Sugimoto K, Nakatochi M, Suguro M, Hirakawa A,
{\bf Hocking TD}, Takeuchi I, Tokunaga T, Takagi Y, Sakamoto A, Aoki T, Naoe
T, Nakamura S, Hayakawa F, Seto M, Tomita A, Kiyoi H. Development and
analysis of patient-derived xenograft mouse models in intravascular
large B-cell lymphoma. {\it Leukemia} 2016.

Chicard M, Boyault S, Colmet-Daage L, Richer W, Gentien D, Pierron G,
Lapouble E, Bellini A, Clement N, Iacono I, Br√©jon S, Carrere M, Reyes
C, {\bf Hocking TD}, Bernard V, Peuchmaur M, Corradini N, Faure-Conter
C, Coze C, Plantaz D, Defachelles A-S, Thebaud E, Gambart M, Millot F,
Valteau-Couanet D, Michon J, Puisieux A, Delattre O, Combaret V,
Schleiermacher G. Genomic copy number profiling using circulating free
tumor DNA highlights heterogeneity in neuroblastoma. {\it Clinical Cancer
Research} 2016.

Maidstone R, {\bf Hocking TD}, Rigaill G, Fearnhead P. On optimal
multiple changepoint algorithms for large data. {\it Statistics and
Computing} (2016). doi:10.1007/s11222-016-9636-3 

{\bf Hocking TD}, Rigaill G, Bourque G. PeakSeg: constrained optimal
segmentation and supervised penalty learning for peak detection in
count data. {\it International Conference on Machine Learning (ICML)},
2015.

Suguro M, Yoshida N, Umino A, Kato H, Tagawa H, Nakagawa M, Fukuhara
N, Karnan S, Takeuchi I, {\bf Hocking TD}, Arita K, Karube K, Tsuzuki
S, Nakamura S, Kinoshita T, Seto M. Clonal heterogeneity of lymphoid
malignancies correlates with poor prognosis. {\it Cancer Sci.} 2014
Jul;105(7):897-904.

{\bf Hocking TD}, Boeva V, Rigaill G, Schleiermacher G,
Janoueix-Lerosey I, Delattre O, Richer W, Bourdeaut F, Suguro M, Seto
M, Bach F, Vert J-P. SegAnnDB: interactive Web-based genomic
segmentation. {\it Bioinformatics} (2014) 30 (11):
1539-1546. DOI:10.1093/bioinformatics/btu072

{\bf Hocking TD}, Wutzler T, Ponting K and Grosjean P. Sustainable,
extensible documentation generation using inlinedocs. {\it Journal of
Statistical Software} (2013), 54(6), 1-20. DOI:10.18637/jss.v054.i06

{\bf Hocking TD}, Schleiermacher G, Janoueix-Lerosey I, Boeva V, Cappo
J, Delattre O, Bach F, Vert J-P. Learning smoothing models of copy
number profiles using breakpoint annotations. {\it BMC Bioinfo.} 2013,
14:164. DOI:10.1186/1471-2105-14-164

{\bf Hocking TD}, Rigaill G, Bach F, Vert J-P. Learning sparse
penalties for change-point detection using max-margin interval
regression. {\it International Conference on Machine Learning (ICML)}, 2013.

{\bf Hocking TD}, Joulin A, Bach F, Vert J-P. Clusterpath: an
Algorithm for Clustering using Convex Fusion Penalties. {\it International Conference on Machine Learning (ICML)}, 2011.

Gautier M, {\bf Hocking TD}, Foulley JL. A Bayesian outlier criterion
to detect SNPs under selection in large data sets. {\it PloS ONE} 5
(8), e11913 (2010).

Doyon Y, McCammon JM, Miller JC, Faraji F, Ngo C, Katibah GE, Amora R,
{\bf Hocking TD}, Zhang L, Rebar EJ, Gregory PD, Urnov FD, Amacher
SL. Heritable targeted gene disruption in zebrafish using designed
zinc-finger nucleases. {\it Nature biotechnology} 26 (6), 702-70
(2008).

\section{\sc Books, Chapters, Manuals}

{\bf Hocking TD} and Killick R. {\it Changepoint detection algorithms
  and applications in R}. Textbook in preparation.

{\bf Hocking TD}. Introduction to Machine Learning and Neural
Networks. Chapter in textbook {\it New Advances in Land Carbon Cycle
  Modeling} edited by Yiqi Luo. (in preparation)

{\bf Hocking TD}. Animated interactive data visualization using the
grammar of graphics (The animint2 Manual), 17 web pages/chapters with
interactive graphics and exercises. (2018)

\section{\sc Conference Tutorials}

{\bf Hocking TD}, Killick R. Introduction to optimal changepoint
detection algorithms, {\it useR} 2017.

{\bf Hocking TD}, Ekstr\o m CT. Understanding and creating interactive
graphics, {\it useR} 2016.

\section{\sc Invited talks (selected)}

University of Waterloo, Universit√© de Montr√©al, Sainte-Justine
Children's Hospital, University of Qu√©bec √† Montr√©al, Polytechnique
Montr√©al (2017); Universit\'e Laval Centre for Big Data Research
(2016); McGill Barbados epigenomics workshop (2015); Sapporo Japan
Workshop on Machine Learning and Applications to Biology (2013);
Google Research New York, Universit\'e Rennes,
Universit\'e Angers, INRIA Lille  (2012); Institut de Biologie
de Lille (2011).
\newpage
\section{\sc Teaching}

Fall 2020, Northern Arizona University, CS499/599, Unsupervised
Learning.

Summer 2020, 90 minute lecture ``Introduction to Machine Learning and
Neural Networks'' for summer school on ``New Advances in Land Carbon
Cycle Modeling.''

Spring 2020, Northern Arizona University, CS499, Deep Learning.

Fall 2019, Northern Arizona University, CS/EE599, Reproducible Machine
Learning Research.

Spring 2019, Northern Arizona University, CS499, Optimization
algorithms for machine learning.

\section{\sc Professional Service}

President of organizing committee for ``R in Montreal 2018'' conference.

Co-administrator and mentor for R project in Google Summer of Code,
since 2012.

Reviewer since 2010: Technometrics, International Conference on
Machine Learning (ICML), Advances in Neural Information Processing
Systems (NIPS), Journal of Machine Learning Research (JMLR),
Artificial Intelligence Review, Journal of Computational and Graphical
Statistics (JCGS), R Journal, Bioinformatics, PLOS Computational
Biology, BMC Bioinformatics, IEEE Transactions on Pattern Analysis and
Machine Intelligence, Information and Inference.

Editor for Journal of Statistical Software since 2018.

\section{\sc Software Online (Selected)} 

Numerous free/open-source software contributions using R, C, C++,
Python, and JavaScript. %\url{https://github.com/tdhock/}

{\bf R}: contributions to base R regex functionalty, ggplot
functionality in plotly package, data reshaping in data.table
package. Maintainer of numerous R packages for machine learning
(changepoint detection, regression, ranking, etc), directlabels for
labeled figures, animint2 for animated interactive figures, inlinedocs
for documentation generation.

{\bf Python}: contributions to pandas module for data manipulation
(str.extractall regex functionality), maintainer of GUI software for
labeling and changepoint detection in genomic data (annotate\_regions,
SegAnnDB).

\begin{comment}
\pagebreak

\section{\sc References}

\noindent {\bf Francis Bach}\\
Scientific Leader,
SIERRA Project-Team\\
%Department of Computer Science\\
Institut National de Recherche en Informatique et en Automatique\\
%Phone: (604)822-9878\\
E-mail: francis.bach@ens.fr\\
Webpage: \url{http://www.di.ens.fr/~fbach}

\noindent {\bf Michael Friedlander}\\
Associate Professor,
Department of Computer Science\\
University of British Columbia\\
%Phone: (604)822-9878\\
E-mail: mpf@cs.ubc.ca\\
Webpage: http://www.cs.ubc.ca/$\sim$mpf

\noindent {\bf Kevin Murphy}\\
Research Scientist,
%Department of Computer Science, Department of Statistics\\
%University of British Columbia\\
Google\\
%Phone: (604)822-9878\\
E-mail: murphyk@gmail.com\\
Webpage: http://www.cs.ubc.ca/$\sim$murphyk


%\noindent {\bf R\'{o}mer Rosales}\\
%Research Scientist\\
%CAD and Knowledge Systems\\
%Siemens Medial Solutions USA, Inc.\\
%%Phone: (610)448 4793\\
%E-mail: romer.rosales@gmail.com\\
%Webpage: http://people.csail.mit.edu/romer/\\
%
%\noindent {\bf Glenn Fung}\\
%Research Scientist\\
%CAD and Knowledge Systems\\
%Siemens Medial Solutions USA, Inc.\\
%E-mail: glenn.fung@siemens.com\\
%Webpage: http://pages.cs.wisc.edu/$\sim$gfung/\\
%
\noindent {\bf Russell Greiner}\\
 Professor, Department of Computing Science\\
University of Alberta\\
%Phone: (604)822-9878\\
E-mail: greiner@cs.ualberta.ca\\
Webpage: http://webdocs.cs.ualberta.ca/$\sim$greiner/
\end{comment}
\end{resume}
\end{document}




